src/awsSqsQueueMonitor.ts

```diff
@@ -0,0 +1,106 @@
+import { EventEmitter } from 'events';
+import { default as AWS } from 'aws-sdk';
+
+export class AwsSqsQueueMonitor extends EventEmitter {
+  private queue: string;
+  private buckets: string[];
+  private monitoring: boolean;
+  private sqs: AWS.SQS;
+
+  constructor(queueUrl: string, buckets: string[] = ['*']) {
+    super();
+
+    this.queue = queueUrl;
+    this.buckets = buckets;
+    this.monitoring = false;
+
+    this.sqs = new AWS.SQS();
+  }
+
+  async poll() {
```

Comment:
This method seems to be doing quite a few things. Maybe it breaking it down into multiple methods would make it easier to read

Reply:
Agreed. I'm likely to tweak this a bit in the next task, I can address this then.